### PR TITLE
Display error messages for Xero Import page

### PR DESCRIPTION
**Pull request by SzymczakJ**
### Details

This PR adds error messages for Xero Import page.
### Fixed Issues

$ https://github.com/Expensify/App/issues/41843
PROPOSAL: N/A

### Tests

1. With a Workspace with a Xero connection
2. Turn on offline mode on troubleshoot page
3. Go to Import page
4. Turn on/off some of the import page items
5. Verify that the XeroImportPage looks like this(there's fade on changed items):
Screenshot 2024-05-13 at 17 02 53

6. Go into one of the XeroImportPage subpages that got changed and Verify that it looks like this:
Screenshot 2024-05-13 at 17 03 06

7. Turn on simulate failing network requests on troubleshoot page and turn off offline mode
8. Go to Import page and turn on/off some of the import page items
9. Verify that the XeroImportPage looks like this(there's red brick road indicator):
Screenshot 2024-05-13 at 17 06 19

10. Go into one of the XeroImportPage subpages that got changed and Verify that it looks like this:
Screenshot 2024-05-13 at 17 06 34

### QA Steps

Same as tests

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/88395093/9fb5dd0e-4d5a-423d-97fe-640ee29363a0

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/88395093/b1f39b11-61b7-46de-908b-a9381c2abd2e

</details>

<details>
<summary>iOS: Native</summary>
https://github.com/Expensify/App/assets/88395093/8d1066f0-b09d-46c6-99ea-98d698018efa

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/88395093/fc26562d-61ca-4fd9-84d7-d91a0dc6f40c

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/88395093/fecf8e39-15d4-4b83-9d74-f07d56e71076

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/88395093/cfe78beb-00e1-4869-9717-dadd98446fad

</details>

### CODE REVIEW

**Comment by lakchote** [comment]
@Expensify/design could you please review these error messages if they look good to you?

**Comment by shawnborton** [comment]
Before we merge this, should we wait until @narefyev91's [PR here](https://github.com/Expensify/App/pull/41670) is merged? That will give us the updated error style that we want.

**Comment by lakchote** [comment]
@trjExpensify this PR will add error messages to the main `Import` page.

@SzymczakJ asked me if we should add error messages to the sub page of the `Import` page also?
It'd look like this below:
image

Could we have your opinion on this?

**Comment by shawnborton** [comment]
That error style seems way off btw - if we do go with an error on this page, it should stretch to the full width of that view and sit underneath the toggle row:
image

**Comment by trjExpensify** [comment]
>@SzymczakJ asked me if we should add error messages to the sub page of the Import page also?
It'd look like this below:

I think there's typically only one place the error message is and that's at the end of the brick road. So if that's on the Tracking Categories page because of a problem with the import toggle, it should only be there. The "red dots" of the red brick road lead to it.

**Comment by SzymczakJ** [comment]
@tjr so you suggest that we shouldn't show error messages on Xero Import Page and instead of that show error messages on Import Page subpages(and show only red dot on Import page). 
It would look something like that:
image
image

cc @trjExpensify @lakchote @shawnborton

**Comment by trjExpensify** [comment]
> @trjExpensify so you suggest that we shouldn't show error messages on Xero Import Page and instead of that show error messages on Import Page subpages(and show only red dot on Import page). It would look something like that: image image
> 
> cc @trjExpensify @lakchote @shawnborton

Yes exactly, because the end of the red brick road is one page deeper and the end of the road is where the error message appears. 👍

**Comment by SzymczakJ** [comment]
I've addressed your review comments @Expensify/design, could you please review this? 🙏
Current version of screens is in the test steps
 cc @shawnborton @trjExpensify

**Comment by shawnborton** [comment]
Cool, I think that feels pretty good to me.

File: src/pages/workspace/accounting/xero/XeroTaxesConfigurationPage.tsx [hunk 1]
```diff
@@ -1,59 +1,42 @@
...
+import ToggleSettingOptionRow from '@pages/workspace/workflows/ToggleSettingsOptionRow';
+import * as Policy from '@userActions/Policy';
 import CONST from '@src/CONST';
 
 function XeroTaxesConfigurationPage({policy}: WithPolicyProps) {
     const {translate} = useLocalize();
     const styles = useThemeStyles();
     const policyID = policy?.id ?? '';
-    const {importTaxRates, pendingFields} = policy?.connections?.xero?.config ?? {};
+    const xeroConfig = policy?.connections?.xero?.config;
```
Comment (filip-solecki) [inline src/pages/workspace/accounting/xero/XeroTaxesConfigurationPage.tsx:17]:
```suggestion
    const xeroConfig = policy?.connections?.xero?.config ?? {};
```
And you don't have to add empty object in each place later

File: src/pages/workspace/accounting/xero/XeroTaxesConfigurationPage.tsx [hunk 2]
```diff
@@ -1,59 +1,42 @@
...
     return (
-        <AccessOrNotFoundWrapper
+        <ConnectionLayout
+            displayName={XeroTaxesConfigurationPage.displayName}
+            headerTitle="workspace.accounting.taxes"
+            title="workspace.xero.taxesDescription"
             accessVariants={[CONST.POLICY.ACCESS_VARIANTS.ADMIN]}
             policyID={policyID}
             featureName={CONST.POLICY.MORE_FEATURES.ARE_CONNECTIONS_ENABLED}
+            contentContainerStyle={[[styles.pb2, styles.ph5]]}
```
Comment (filip-solecki) [inline src/pages/workspace/accounting/xero/XeroTaxesConfigurationPage.tsx:29]:
```suggestion
            contentContainerStyle={[styles.pb2, styles.ph5]}
```
Should be enough

File: src/pages/workspace/accounting/xero/XeroTrackingCategoryConfigurationPage.tsx [hunk 3]
```diff
@@ -59,29 +61,16 @@ function XeroTrackingCategoryConfigurationPage({policy}: WithPolicyProps) {
...
-                                }
-                            />
-                        </View>
-                    </OfflineWithFeedback>
-                </View>
-            </View>
+            <ToggleSettingOptionRow
+                title={translate('workspace.accounting.import')}
+                switchAccessibilityLabel={translate('workspace.xero.trackingCategories')}
+                isActive={!!isSwitchOn}
```
Comment (filip-solecki) [inline src/pages/workspace/accounting/xero/XeroTrackingCategoryConfigurationPage.tsx:67]: There is no need to use `!!` as it is Booelan already

File: src/pages/workspace/accounting/xero/XeroTrackingCategoryConfigurationPage.tsx [hunk 4]
```diff
@@ -22,7 +21,10 @@ function XeroTrackingCategoryConfigurationPage({policy}: WithPolicyProps) {
     const {translate} = useLocalize();
     const styles = useThemeStyles();
     const policyID = policy?.id ?? '';
-    const {importTrackingCategories, pendingFields} = policy?.connections?.xero?.config ?? {};
+    const xeroConfig = policy?.connections?.xero?.config;
+    const {importTrackingCategories} = policy?.connections?.xero?.config ?? {};
+
+    const isSwitchOn = Boolean(importTrackingCategories);
```
Comment (filip-solecki) [inline src/pages/workspace/accounting/xero/XeroTrackingCategoryConfigurationPage.tsx:27]:
```suggestion
    const isSwitchOn = !!importTrackingCategories;
```

File: src/pages/workspace/accounting/xero/XeroTrackingCategoryConfigurationPage.tsx [hunk 5]
```diff
@@ -22,7 +21,10 @@ function XeroTrackingCategoryConfigurationPage({policy}: WithPolicyProps) {
     const {translate} = useLocalize();
     const styles = useThemeStyles();
     const policyID = policy?.id ?? '';
-    const {importTrackingCategories, pendingFields} = policy?.connections?.xero?.config ?? {};
+    const xeroConfig = policy?.connections?.xero?.config;
```
Comment (filip-solecki) [inline src/pages/workspace/accounting/xero/XeroTrackingCategoryConfigurationPage.tsx:24]: same suggestion as above

File: src/pages/workspace/accounting/xero/import/XeroCustomerConfigurationPage.tsx [hunk 6]
```diff
@@ -31,32 +30,22 @@ function XeroCustomerConfigurationPage({policy}: WithPolicyProps) {
...
+                switchAccessibilityLabel={translate('workspace.xero.customers')}
+                subMenuItems={
+                    <MenuItemWithTopDescription
+                        interactive={false}
+                        title={translate('workspace.common.tags')}
+                        description={translate('workspace.qbo.displayedAs')}
+                        wrapperStyle={styles.sectionMenuItemTopDescription}
+                    />
+                }
+                isActive={!!isSwitchOn}
```
Comment (filip-solecki) [inline src/pages/workspace/accounting/xero/import/XeroCustomerConfigurationPage.tsx:44]: same as above

File: src/pages/workspace/categories/WorkspaceCategoriesSettingsPage.tsx [hunk 7]
```diff
@@ -23,6 +23,7 @@ function WorkspaceCategoriesSettingsPage({policy, route}: WorkspaceCategoriesSet
     const isConnectedToAccounting = Object.keys(policy?.connections ?? {}).length > 0;
     const policyID = route.params.policyID ?? '';
     const [policyCategories] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${policyID}`);
+    const toggleSubtitle = isConnectedToAccounting ? `${translate('workspace.categories.needCategoryForExportToIntegration')} ${translate('workspace.accounting.qbo')}` : '';
```
Comment (filip-solecki) [inline src/pages/workspace/categories/WorkspaceCategoriesSettingsPage.tsx:26]: is it correct? We check here for any connection(may be xero) and display qbo text

**Review by filip-solecki — APPROVED** [review]
LGTM

**Review by fedirjh — APPROVED** [review]

**Review by lakchote — APPROVED** [review]

**Comment by Julesssss** [comment]
FYI, a [bug](https://github.com/Expensify/App/issues/42268) was raised where the RBR error isn't always displayed. As this is a new feature I'm not marking it as a blocker though.

File: src/pages/workspace/categories/WorkspaceCategoriesSettingsPage.tsx [hunk 8]
```diff
@@ -44,7 +53,9 @@ function WorkspaceCategoriesSettingsPage({policy, route}: WorkspaceCategoriesSet
                 <View style={styles.flexGrow1}>
                     <ToggleSettingOptionRow
                         title={translate('workspace.categories.requiresCategory')}
-                        subtitle={isConnectedToAccounting ? `${translate('workspace.categories.needCategoryForExportToIntegration')} ${translate('workspace.accounting.qbo')}` : ''}
+                        titleStyle={styles.textStrong}
```
Comment (hayata-suenaga) [inline src/pages/workspace/categories/WorkspaceCategoriesSettingsPage.tsx:56]: [The text in the `ToggleSettingOptionRow` shouldn't be bold.](https://expensify.slack.com/archives/C01SKUP7QR0/p1715875595366199) 🙇

**Comment by hayata-suenaga** [comment]
> tl:dr I think the bold text was there before my PR got merged, but correct me if I'm wrong 🙇

that's fair I think the timing of removing the bold style and this PR probably coincided 😅 Anyway, thank you for your work @SzymczakJ!

**Pull request merged**